src/app/http.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { HttpClient} from '@angular/common/http';
+import { Links } from './links';
+// import { map } from 'rxjs/operators';
+import { Observable, of, from } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HttpService {
```

Comment:
The name of the service is to abstract
Please rename it to ChartDataService

Reply:
fixed